ostd/src/task/mod.rs

```diff
@@ -166,7 +166,7 @@ impl TaskOptions {
     }
 
     /// Builds a new task without running it immediately.
-    pub fn build(self) -> Result<Arc<Task>> {
+    pub fn build(self) -> Result<Task> {
```

Comment:
Could you explain the rationale of this change?

Reply:
This is because I need to use `Arc::new_cyclic` to pass `weak_task` to Thread when creating a new thread
```rust
fn create_new_kernel_task(....) -> Arc<Task>{
    Arc::new_cyclic(|weak_task| {
        let thread = Thread::new(weak_task.clone(), .......);
        TaskOptions::new(....).data(thread)
    })
}
```